### ISSUE TITLE
Error handling of regular connect errors causes fatal panic

### ISSUE THREAD

**Issue by xkisu** [eclipse/paho.golang#74]
**Describe the bug**
When an error result code is returned from the server on `client.Connect()` the Paho client incorrectly tries to call `PingHandler.Stop()` (via client.go#141) - however the ping handler is not started until `client.go#215` so this results in a "close of nil channel" error.

**To reproduce**
Connect to a MQTT broker that returns an error result code on Connect, such as `DisconnectNotAuthorized`.

```golang
// Create the MQTT client for testing
client := paho.NewClient()
client.Conn = tcpConn

// Send a connection packet with invalid auth.
//
// This results in a fatal panic due to the PingHandler not being
// started, but the error tries to stop it and close it's channel
// anyways.
_, err := client.Connect(ctx, &paho.Connect{
	Password:     []byte("password1234"),
	PasswordFlag: true,
	Username:     "notvalid",
	UsernameFlag: true,
})
```

**Debug output**
Stack trace:
```
runtime.fatalpanic at panic.go:1190
runtime.gopanic at panic.go:1065
runtime.closechan at chan.go:357
github.com/eclipse/paho.golang/paho.(*PingHandler).Stop at pinger.go:91
github.com/eclipse/paho.golang/paho.(*Client).Error at client.go:345
github.com/eclipse/paho.golang/paho.(*Client).Incoming at client.go:235
github.com/eclipse/paho.golang/paho.(*Client).Connect.func1 at client.go:141
runtime.goexit at asm_amd64.s:1371
 - Async stack trace
github.com/eclipse/paho.golang/paho.(*Client).Connect at client.go:139
```

**Expected behaviour**
The Connect method should return the server result code as an error and not panic.

**Software used:**
Golang 1.16
paho.golang v0.9.0

**Comment by fracasula** [comment]
Hi @xkisu I suggest you try with `master` again, @alsm has not created a release for quite a while and the code has changed a lot since then.

**Comment by fracasula** [comment]
@alsm is something preventing us for doing at least a `v0.10.0` with the latest?

**Comment by xkisu** [comment]
> Hi @xkisu I suggest you try again with `master`. @alsm has not created a release for quite a while and the code has changed a lot since then.

Ah, thank you! It works correctly with master - that definitely needs to be included in a new release since `v0.9.0` is currently what go modules will grab by default.

**Comment by alsm** [comment]
New release is out, sorry about that
